packages/board-server/Dockerfile

```diff
@@ -1,11 +1,34 @@
-FROM node:20
-COPY --from=breadboard . /breadboard/
-# For the time being, we will assume container deployments are exclusively sqlite.
-ENV STORAGE_BACKEND "sqlite"
-ENV GOOGLE_CLOUD_PROJECT "n/a"
-ENV ALLOWED_ORIGINS ""
-WORKDIR /breadboard
+# Build stage
+FROM node:20 AS build
+WORKDIR /build
+
+# Copy the entire monorepo
+COPY --from=breadboard / ./
+
+# Install ALL dependencies, including devDependencies
 RUN npm ci
-WORKDIR /breadboard/packages/board-server
+
+# Build the project
+WORKDIR /build/packages/board-server
 RUN npm run build
-CMD ["npm", "run", "serve", "--", "--host=0.0.0.0"]
\ No newline at end of file
+
+# Production stage
+FROM node:20-slim
+WORKDIR /app
+ENV NODE_ENV=production
+ENV STORAGE_BACKEND="sqlite"
+ENV GOOGLE_CLOUD_PROJECT="n/a"
+ENV ALLOWED_ORIGINS=""
+
+# Copy necessary files from the build stage
+COPY --from=build /build/packages/board-server/dist ./dist
+COPY --from=build /build/packages/board-server/src ./src
+COPY --from=build /build/packages/board-server/package.json ./
+COPY --from=build /build/packages/board-server/public ./public
+
+# Install production dependencies and tsx
+RUN npm install --only=production && \
+    npm install -g tsx
+
+EXPOSE 3000
+CMD ["node", "dist/server/index.js"]
```

Comment:
Don't need to set the host here like before?

Reply:
Can confirm it also works passing --host=0.0.0.0, so I'll include it.